scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,367 @@
+      two values.
+
+    Left-, right-, and interval-censored data can be represented by
+    `CensoredData`.
+
+    For interval-censored data, pass the lower and upper limits of
+    the intervals as one-dimensional arrays to the `CensoredData`
+    constuctor.  By using the values `np.inf` and `-np.inf`, left-
+    and right-censored data may also be created with the constructor.
+
+    For convenience, the class methods ``left_censored`` and
+    ``right_censored`` are provided to create a `CensoredData`
+    instance from a single one-dimensional array of measurements
+    and a corresponding boolean array to indicate which measurements
+    are censored.
+
+    Parameters
+    ----------
+    x : array_like, 1D
+        Uncensored observations.
+    left : array_like, 1D
+        Left-censored observations.
+    right : array_like, 1D
+        Right-censored observations.
+    intervals : array_like, 2D, with shape (m, 2)
+        Interval-censored observations.  Each row ``intervals[k, :]``
+        represents the interval for the kth interval-censored observation.
+
+    Notes
+    -----
+    In the input array `intervals`, the lower bound of the interval may
+    be ``-inf``, and the upper bound may be ``inf``, but at least one must be
+    finite). When the lower bound is ``-inf``, the row represents a left
+    censored observation, and when the upper bound is ``inf``, the row
+    represents a right-censored observation.  If the length of an interval
+    is 0 (i.e. ``intervals[k, 0] == intervals[k, 1]``, the observation is
+    treated as uncensored.  Consequently, one can represent all the types
+    of censored and uncensored data in ``intervals``, but it is generally
+    more convenient to use `x`, `left` and `right` for uncensored,
+    left-censored and right-censored observations, respectively.
+
+    Examples
+    --------
+    In the most general case, a censored data set may contain values that
+    are left-censored, right-censored, interval-censored, and uncensored.
+    For example, here we create a data set with five observations.  Two
+    are uncensored (values 1 and 1.5), one is a left-censored observation
+    of 0, one is a right-censored observation of 10 and one is
+    interval-censored in the interval [2, 3].
+
+    >>> from scipy.stats import CensoredData
+    >>> data = CensoredData(x=[1, 1.5], left=[0], right=[10],
+    ...                     intervals=[[2, 3]])
+    >>> print(data)
+    CensoredData(5 values: 2 not censored, 1 left-censored,
+    1 right-censored, 1 interval-censored)
+
+    Equivalently,
+    >>> data = CensoredData(intervals=[[1, 1],
+    ...                                [1.5, 1.5],
+    ...                                [-np.inf, 0],
+    ...                                [10, np.inf],
+    ...                                [2, 3]])
+    >>> print(data)
+    CensoredData(5 values: 2 not censored, 1 left-censored,
+    1 right-censored, 1 interval-censored)
```

Comment:
Is the distinction about `interval-censored` useful to make here? In the end it will fall into the other categories and would provide more info about the actual censoring.

Reply:
Can you clarify the question? I added this example.
Is it about why this part of the example is needed, or is it why "1 interval-censored" is included in the string representation?

Reply:
Ooops no my bad here, I got confused for some reason.